lib/libc/common/Kconfig

```diff
@@ -8,26 +8,30 @@ config COMMON_LIBC_ABORT
 
 config COMMON_LIBC_ASCTIME
 	bool
+	depends on !PICOLIBC
 	help
 	  common implementation of asctime().
 
 config COMMON_LIBC_ASCTIME_R
 	bool "Thread-safe version of asctime()"
 	default y if POSIX_THREAD_SAFE_FUNCTIONS
```

Comment:
These lines should be removed, and instead in a POSIX Kconfig, we should use a pattern like:

```
config POSIX_C_LANG_SUPPORT_R
    select LIBC_COMMON_ASCTIME_R if ! TC_PROVIDES_POSIX_C_LANG_SUPPORT_R
    ...
```

That way picolibc could just select `TC_PROVIDES_POSIX_C_LANG_SUPPORT_R` and cover all of those functions at once. That might be closer in intent to the way that the TC_PROVIDES macros gate files being built in cmakelists.txt and we don't need to use `depends on ! PICOLIBC`

Reply:
Had to remove a couple of redundant 'default y if POSIX_THREAD_SAFE_FUNCTIONS' clauses. Looks like a cleaner plan, thanks!